documentation/internal/Filters-for-Applications.md

```diff
@@ -0,0 +1,292 @@
+
+Options:
+
+- Set of `outcome` common for all templates (application types)
+
+##### action
+
+Example: `action=edit-draft`
+Considered in implementation: **Yes** [#251 - Filters]
+Can have combined values: **Yes** e.g: `action=edit-draft,`
+
+Options:
+
+- Applicant:
+  `edit-draft`
+  `make-updates`
+  `renew`
+  `view` (Submitted)
+- Reviewer
+  `start-review`
+  `continue-review`
+  `review-updates`
+  `view` (Submitted)
+- Consolidator
+  `assign`
+  `consolidate`
+  `review-updates`
+  `view` (Submitted)
+
+---
+
+#### String filters:
+
+##### assigned
+
+Example: `assigned=none`
+Considered in implementation: **Yes** [#251 - Filters]
+Can have combined values: **Yes** e.g: `assigned=none,"Carl"`
+
+Options:
+
+- `none` - Applications with sections not-assigned
+- `"username"` - Reviewer username
+
+##### consolidator
+
+Example: `consolidator="Tony"`
+Considered in implementation: **Yes** [#251 - Filters]
+Can have combined values: **Yes** e.g: `consolidator="Tony","Andrei"`
+
+Options:
+
+- `"username"` - Consolidator username
+
+##### applicant
+
+Example: `applicant="John"`
+Considered in implementation: **No**
+Can have combined values: **Yes** e.g: `applicant="John","Nicole"`
+
+Options:
+
+- `"username"` - Applicant username
+
+##### org
+
+Example: `org="Company A"`
+Considered in implementation: **Yes** [#251 - Filters]
+Can have combined values: **Yes** e.g: `org="Company A","Company B"`
+
+Options:
+
+- `"organisation name"`
+
+##### search
+
+Example: `search="abc 123"`
+Considered in implementation: **Yes** [#251 - Filters]
+Can have combined values: **No**
+
+Options:
+
+- String containing [A-Z], [a-z], [0-9], _space_
+- Uses _starts with_ on search including following columns: "Application name", "Stage", ... (**TODO: continue listing columns**)
+- (See more rules on Formats & Rules)
+
+---
+
+#### Date filters:
+
+##### last-active-date
+
+Example: `last-active-date=2021-01-01`
+Considered in implementation: **Yes** [#228 - Dates in columns]
+Can have combined values: **Yes** e.g: `last-active-date=today,last-week`
+
+Options:
+
+- Pre-defined string: `today` (See more formats on Formats & Rules)
+- Single date: `YYYY-MM-DD`
+- Period (start:end): `YYYY-MM-DD:YYYY-MM-DD`
+
+##### deadline-date
+
+Example: `deadline-date=2021-01-31`
+Considered in implementation: **Yes** [#228 - Dates in columns]
+Can have combined values: **Yes** e.g: `deadline-date=today,2021-01-02`
+
+Options:
+
+- Pre-defined string: `today` (See more formats on Formats & Rules)
+- Single date: `YYYY-MM-DD`
+- Period (start:end): `YYYY-MM-DD:YYYY-MM-DD`
+
+---
+
+#### Integer filters:
+
+##### page
+
+Example: `page=1`
+Considered in implementation: **Yes** [#210 - Pagination]
+Can have combined values: **No**
+
+Options:
+
+- Number between 1 - N (maximum of pages?)
+
+##### per-page
+
+Example: `per-page=20`
+Considered in implementation: **Yes** [#210 - Pagination]
+Can have combined values: **No**
+
+Options:
+
+- Number between 10 - 100
```

Comment:
Why enforce a limit at all? I mean, we'd provide some "pre-set" options in the pagination controller, but there's no reason why the query interpretor has to have limits.

Reply:
Ok. removing the limits